datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -52,131 +55,311 @@ fn sign_extend_be(b: &[u8]) -> [u8; 16] {
-        match $column_statistics {
-            ParquetStatistics::Boolean(s) => Some(ScalarValue::Boolean(Some(*s.$func()))),
-            ParquetStatistics::Int32(s) => {
-                match $target_arrow_type {
-                    // int32 to decimal with the precision and scale
-                    Some(DataType::Decimal128(precision, scale)) => {
-                        Some(ScalarValue::Decimal128(
-                            Some(*s.$func() as i128),
-                            *precision,
-                            *scale,
-                        ))
-                    }
-                    Some(DataType::Int8) => {
-                        Some(ScalarValue::Int8(Some((*s.$func()).try_into().unwrap())))
-                    }
-                    Some(DataType::Int16) => {
-                        Some(ScalarValue::Int16(Some((*s.$func()).try_into().unwrap())))
-                    }
-                    Some(DataType::UInt8) => {
-                        Some(ScalarValue::UInt8(Some((*s.$func()).try_into().unwrap())))
-                    }
-                    Some(DataType::UInt16) => {
-                        Some(ScalarValue::UInt16(Some((*s.$func()).try_into().unwrap())))
-                    }
-                    Some(DataType::UInt32) => {
-                        Some(ScalarValue::UInt32(Some((*s.$func()) as u32)))
-                    }
-                    Some(DataType::Date32) => {
-                        Some(ScalarValue::Date32(Some(*s.$func())))
-                    }
-                    Some(DataType::Date64) => {
-                        Some(ScalarValue::Date64(Some(i64::from(*s.$func()) * 24 * 60 * 60 * 1000)))
-                    }
-                    _ => Some(ScalarValue::Int32(Some(*s.$func()))),
-                }
+macro_rules! get_statistics_iter {
+    ($column_statistics_iter:expr, $func:ident, $bytes_func:ident, $target_arrow_type:expr) => {{
+        match $target_arrow_type {
+            Some(DataType::Boolean) => Some(Arc::new(BooleanArray::from_iter(
+                $column_statistics_iter.map(|statistic| {
+                    statistic.and_then(|x| {
+                        if !x.has_min_max_set() {
+                            return None;
+                        }
+                        match x {
+                            ParquetStatistics::Boolean(s) => Some(*s.$func()),
+                            _ => None,
+                        }
+                    })
+                }),
+            )) as ArrayRef),
+            Some(DataType::Int8) => Some(Arc::new(Int8Array::from_iter(
+                $column_statistics_iter.map(|statistic| {
+                    statistic.and_then(|x| {
+                        if !x.has_min_max_set() {
+                            return None;
+                        }
+                        match x {
+                            ParquetStatistics::Int32(s) => {
+                                Some((*s.$func()).try_into().unwrap())
```

Comment:
I think these unwraps are not great (though I see you didn't add them) -- as they may panic on invalid parquet statistics